src/js/models/panel_model.js

```diff
@@ -243,13 +243,26 @@
             var oldLabs = this.get("labels");
             // Need to clone the list of labels...
             var labs = [];
+            var newLabKey = [];
+            for (var i=0; i<labels.length; i++) {
+                newLabKey.push(this.get_label_key(labels[i]));
+            }
+
             for (var i=0; i<oldLabs.length; i++) {
-                labs.push( $.extend(true, {}, oldLabs[i]) );
+                lbl = oldLabs[i];
+                lbl_key = this.get_label_key(lbl);
+                // for existing label that matches...
+                if (!newLabKey.includes(lbl_key)) {
+                    // otherwise leave un-edited
+                    labs.push( $.extend(true, {}, lbl));
+                }
```

Comment:
If I add a new label and it already exists, it removes the existing one and adds the new one, which is slightly unexpected. E.g. If I add labels, `1`, `2`, `3`, `4`, and then add another duplicate label `2`, then I now have labels
`1`, `3`, `4`, `2`, whereas I'd expect the attempt to add a duplicate should have no effect.

Reply:
Ok, I have corrected it.